ethers-middleware/src/nonce_manager.rs

```diff
@@ -160,7 +160,12 @@ where
                 tracing::debug!(?nonce, "Sent transaction");
                 let tx_count_for_resync = self.get_tx_count_for_resync();
                 if new_txs_since_resync >= tx_count_for_resync {
-                    let onchain_nonce = self.get_transaction_count(self.address, block).await?;
+                    // subtract 1 because the nonce returned by `get_transaction_count`
+                    // is the next one to be used, whereas we want the last one that landed
+                    let onchain_nonce = self
+                        .get_transaction_count(self.address, block)
+                        .await?
+                        .saturating_sub(1.into());
```

Comment:
there's an edge case here if it's the first ever tx, where get_transaction_count returns 0. Think it's a concern or that it's fine for now?

Reply:
this branch would only be hit after `tx_count_for_resync` txs which is >0 so we're good, but it would be a concern if `tx_count_for_resync` was `0`

Reply:
actually if none of the `tx_count_for_resync` land this would be a problem, but I think it's very unlikely we run into that

Reply:
I'll set it to `min(onchain_nonce, 1)`